### ISSUE TITLE
[FR] A more effective support for the learning rate schedule 

### ISSUE THREAD

**Issue by CSJDeveloper** [TL-System/plato#212]
**Is your feature request related to a problem? Please describe.**
The learning rate (lr) schedule will not work in the current implementation of plato's trainer, i.e., `plato/trainers/basic.py`. The training process will suffer from a constant learning rate. The main reason is that the trainer will create a new lr schedule each round. Then, the epoch within one round will always start from 1. Therefore, if the lr schedule works based on this local epoch, the lr will never be modified correctly as that in the central learning. For example, under settings where the initial learning rate is 0.1, and the local epoch is 5, the learning rate will maintain 0.1 during the whole training process (100 communication rounds) no matter what lr schedule is used.

Anyone can reproduce this potential issue by tracking the learning rate in each round. 

**Describe the solution you'd like**

However, general machine learning follows that the learning rate can be decreased as the training epoch increases. I believe that this is why Plato supports the learning rate schedule in the trainer. 

It is better to run a training process in which the scheduler decays the client's learning rate when the training loss plateaus.

**Describe alternatives you've considered**

Therefore, to introduce this property (a changeable learning rate based on the training status) to Plato's training framework, the simplest way is to define a variable computed as:

 `lr_schedule_base_epoch = (current_round - 1) * epochs`

Then, the defined lr_schedule can be updated as: 

`lr_schedule.step(lr_schedule_base_epoch + epoch)` in the local training stage. 

This lets Plato vary the learning rate as the training (communication round) progresses. 

**Additional context**

I have implemented such property of lr schedule in my branch as a changeable lr is important to my model training and most machine learning methods. Without a changeable learning rate, it is hard to beat the state-of-the-art methods equipped with a dynamic learning rate. I am not sure whether this is necessary for others. But, this is a valuable feature that can be added to Plato.

**Comment by baochunli** [comment]
Supported in PR #225.
